Modify note content about kubectl, CRDs, and subresources

Shubham82 opened the pull request:
This PR updated the `Note` Content for the  `Kubectl Subresource Support KEP` according to the current state under the [Upgrade existing objects to a new stored version](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version) in [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/) task.

Fixes: #34463

sbueringer commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,14 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+subresource on a CRD. See the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
 
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+Here is an example of How to patch the `status` subresource for the CRD using the Kubectl 
+tool:
```
```suggestion
Here is an example of How to patch the `status` subresource for the CRD using `kubectl`:
```

Shubham82 replied:
I have addressed your comment. PTAL!

Shubham82 replied:
addressed

sbueringer reviewed (commented):
Just a suggestion.

sftim commented:
/retitle Modify note content about kubectl, CRDs, and subresources

Shubham82 commented:
> @Shubham82 please go easy on reviewers - you @-mentioned 3 folks there, and this small bugfix PR only needs the one reviewer. The issue that you're fixing only has backlog priority.
> 
> Next time, please consider waiting a day or so before nudging even one person to review.

@sftim Sure, I will take care of it next time.

sbueringer commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,13 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
-The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
+In Kubernetes v1.24, an alpha flag `--subresource` is added to get, patch, edit and replace 
+with kubectl commands to fetch and update `status` and `scale` subresources. Due to this, 
+From Kubernetes v1.24 the `kubectl` tool can now be used to edit or patch the `status` 
```
```suggestion
from Kubernetes v1.24 the `kubectl` tool can now be used to edit or patch the `status` 
```

Shubham82 replied:
addressed

sbueringer commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,13 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+subresource on a CRD. See the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
 
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+Here is an example of How to patch the `status` subresource for the CRD using the `Kubectl`:
```
```suggestion
Here is an example of How to patch the `status` subresource for the CRD using the `kubectl`:
```

Shubham82 replied:
addressed

sbueringer reviewed (commented):
last two nits from my side.

I'm not sure how it works in this repository, but might be good to squash the commits

reylejano commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,13 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+subresource on a CRD. See the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
 
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+Here is an example of How to patch the `status` subresource for the CRD using the `Kubectl`:
```
```suggestion
Here is an example of How to patch the `status` subresource for the CRD using `kubectl`:
```

Shubham82 replied:
addressed

reylejano commented:
After addressing comments, please squash commits into 1 commit
Please see [squashing commits in the Kubernetes documentation](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits)

Shubham82 commented:
> After addressing comments, please squash commits into 1 commit Please see [squashing commits in the Kubernetes documentation](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits)

Sure I will do that.
Thanks.

Shubham82 commented:
> After addressing comments, please squash commits into 1 commit Please see [squashing commits in the Kubernetes documentation](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits)

Done, PTAL

sftim reviewed (changes requested):
This revised text isn't right.

We avoid referring readers to KEPs to find out how implemented features actually work. Instead, we (SIG Docs, plus the responsible SIG) take a different approach: we _document the feature_.

The revised page should jump straight into explaining how to patch a CRDs subresource, and clarify that anyone running Kubernetes older than v1.24 should read the docs for the version of Kubernetes that they're actually running.

sftim commented:
/wg api-expression
/sig api-machinery

Shubham82 commented:
Hi @sftim 
Instead of writing this See the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details. can we write like this: 

> See How to [patch a CRDs subresource](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource#patch) using kubectl. 
> This page is part of the documentation for Kubernetesv{currentVersion}. If you are running a different version of Kubernetes, consult the documentation for that release.

Shubham82 commented:
> Hi @sftim Instead of writing this See the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details. can we write like this:
> 
> > See How to [patch a CRDs subresource](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource#patch) using kubectl.
> > This page is part of the documentation for Kubernetesv{currentVersion}. If you are running a different version of Kubernetes, consult the documentation for that release.

Hi @sftim
A gentle reminder

sftim commented:
We shouldn't link to https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource#patch as that is _still_ a link to a KEP and not a documentation page.

We might need to:
1. Update https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/ to explain subresource patching
2. Link there

BTW @Shubham82 I can't guarantee any response time on advice on how to make changes. If you're stuck on how to move a PR forward, asking in the SIG Docs Slack channel should work better because then there is more than one person who'll see the request for advice.

Shubham82 commented:
> BTW @Shubham82 I can't guarantee any response time on advice on how to make changes. If you're stuck on how to move a PR forward, asking in the SIG Docs Slack channel should work better because then there is more than one person who'll see the request for advice.

Sure @sftim, thanks for advice.

sftim commented:
@Shubham82 what's your intention with this PR? Are you clear on what steps to take to move the work forward?

Shubham82 commented:
> @Shubham82 what's your intention with this PR?

Hi, @sftim sorry for the delay, was busy with other stuff. now will start work on this.

Shubham82 commented:
> Are you clear on what steps to take to move the work forward?

Yes, first will add the section here https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/ about subresource patching (with example) and then link that under Note in [Upgrade existing objects to a new stored version](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version)

will soon update this.

Shubham82 commented:
Hi @sftim
Could please take look into this?

Thanks!

sftim commented:
Hi folks. Any English reviewer is welcome to check this; it doesn't have to be me.

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
 
+## Use subresource flag with kubectl patch command to update the deployment
+
+In kubernetes v1.24, kubectl commands like get, patch, edit and replace will now contain a new flag `--subresource=[subresource-name]` which will allow fetching and updating `status` and `scale` subresources for all resources (built-in and CRs) that support these subresources.
```
I'm not happy with this text. It sounds like it was written before Kubernetes v1.24 was released.
Instead, please describe how the latest Kubernetes release works, right now. Explain that this is generic across any API with a `status` or `scale` subresource, and that the Deployment is an example.

sftim commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,14 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+This page is part of the documentation for Kubernetesv{currentVersion}. If you are running a different version of Kubernetes, consult the documentation for that release.
 
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+Here is an example of How to patch the `status` subresource for the CRD using `kubectl`:
```
```suggestion
Here is an example of how to patch the `status` subresource for the CRD, using `kubectl`:
```

Shubham82 replied:
addressed

sftim commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,14 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
 
 {{< note >}}
-The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
+In Kubernetes v1.24, an alpha flag `--subresource` is added to get, patch, edit and replace 
+with kubectl commands to fetch and update `status` and `scale` subresources. Due to this, 
+from Kubernetes v1.24 the `kubectl` tool can now be used to edit or patch the `status`
+subresource on a CRD. See [How to patch deployment by using subresource flag](/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#Use-subresource-flag-with-kubectl-patch-command-to-update-the-deployment).
```
I'm not happy with this text. It sounds like too much as if it was written before Kubernetes v1.24 was released.
Instead, please describe how the latest Kubernetes release works, right now.

Shubham82 replied:
addressed

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+
+In kubernetes v1.24, kubectl commands like get, patch, edit and replace will now contain a new flag `--subresource=[subresource-name]` which will allow fetching and updating `status` and `scale` subresources for all resources (built-in and CRs) that support these subresources.
+
+Here's the configuration file for a Deployment that has two replicas.
```
(nit)
```suggestion
Here's a manifest for a Deployment that has two replicas:
```

Shubham82 replied:
addressed

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
 And you can see the `retainKeys` strategy in the
 [Kubernetes API documentation](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#deploymentspec-v1-apps).
 
+## Use subresource flag with kubectl patch command to update the deployment
```
I would move this inside:
```markdown
## Alternate forms of the kubectl patch command
```

Also, a text suggestion:

```suggestion
## Update an object's replica count using `kubectl patch` with `--subresource` {#scale-kubectl-patch}

{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
```
(is this right, alpha in v1.24?)

Shubham82 replied:
Yes, it is alpha in v1.24.
Ref: https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/2590-kubectl-subresource/kep.yaml#L24

Shubham82 replied:
addressed

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+subresource-demo-777d58f9d8-fkf4l   1/1     Running   0          28s
+```
+
+Now, Patch your Deployment with `--subresource=[subresource-name]` flag:
````
```suggestion
Now, patch that Deployment with `--subresource=[subresource-name]` flag:
```

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+kubectl patch deployment subresource-demo --subresource='scale' --type='merge' -p '{"spec":{"replicas":3}}'
+```
+
+you can see following output, which means your deployment is successfully patched:
````
```suggestion
The output is:
```

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+View the Pods associated with your patched Deployment:
+
+```shell
+kubectl get pods
````
Use a label selector here to only show the demo Pods.

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+kubectl get pods
+```
+
+In the output, you can see that Deployment has two Pods.
````
```suggestion
In the output, you can see that Deployment has two Pods. For example:
```

Shubham82 replied:
addressed

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+```
+
+```yaml
+spec:
````
```suggestion
...
spec:
```
You could also use jsonpath output to select just the status.

Shubham82 replied:
we can use jsonpath but on using it give output in the below format:

```
$ kubectl get deployment subresource-demo  -o=jsonpath='{.status}'
{"availableReplicas":3,"conditions":[{"lastTransitionTime":"2022-08-17T07:35:17Z","lastUpdateTime":"2022-08-17T07:35:57Z","message":"ReplicaSet \"subresource-demo-777d58f9d8\" has successfully progressed.","reason":"NewReplicaSetAvailable","status":"True","type":"Progressing"},{"lastTransitionTime":"2022-08-18T08:52:25Z","lastUpdateTime":"2022-08-18T08:52:25Z","message":"Deployment has minimum availability.","reason":"MinimumReplicasAvailable","status":"True","type":"Available"}],"observedGeneration":2,"readyReplicas":3,"replicas":3,"updatedReplicas":3}
```

IMO it seems good and also  YAML content is used to show the output in the corresponding docs, so for the uniformity of the docs, we can write in this way.

sftim commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
````diff
@@ -382,6 +382,81 @@ You can also see the `retainKeys` strategy in the [OpenApi spec](https://raw.git
+```
+
+{{< note >}}
+If `--subresource` flag is used for a resource that doesn't support the subresource, a NotFound error will be returned.
````
```suggestion
If you run `kubectl patch` and specify `--subresource` flag for resource that doesn't support that
particular subresource, the API server returns a 404 Not Found error.
```

Shubham82 replied:
addressed

Shubham82 commented:
Thanks, @sftim for your comments, I will update it soon.

Shubham82 commented:
Addressed the comments
PTAL!

divya-mohan0209 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
-The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
+The flag `--subresource` is added to get, patch, edit and replace with kubectl commands to 
+fetch and update `status` and `scale` subresources of all the API resources that support 
+these subresources. It is applicable for kubectl version v1.24 or more. Previously fetching 
```
```suggestion
these subresources. It is applicable for kubectl version v1.24 or higher. Previously fetching 
```

Shubham82 replied:
addressed

divya-mohan0209 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+fetch and update `status` and `scale` subresources of all the API resources that support 
+these subresources. It is applicable for kubectl version v1.24 or more. Previously fetching 
+subresources (like status) of API objects via kubectl involved using `kubectl --raw`, and 
+patching subresources using kubectl is not possible at all and requires using curl directly, 
```
```suggestion
patching subresources using kubectl was not possible at all and required using curl directly.
```

Shubham82 replied:
addressed

divya-mohan0209 commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -417,6 +417,86 @@ kubectl patch deployment patch-demo --patch-file patch-file.json
+
+{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
+
+The flag `--subresource=[subresource-name]` is used with kubectl commands like get, patch, edit and replace to fetch and update `status` and `scale` subresources of the resources (applicable for kubectl version v1.24 or more). This flag is used with all the API resources (built-in and CRs) which has `status` or `scale` subresource, Deployment is one of the examples which supports these subresourc ...
```
```suggestion
The flag `--subresource=[subresource-name]` is used with kubectl commands like get, patch, edit and replace to fetch and update `status` and `scale` subresources of the resources (applicable for kubectl version v1.24 or more). This flag is used with all the API resources (built-in and CRs) which has `status` or `scale` subresource. Deployment is one of the examples which supports these subresources.
```

Shubham82 replied:
addressed

Shubham82 commented:
Thanks, @divya-mohan0209 for taking a look at it.
I will update it as per your comment.

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
 
 {{< note >}}
-The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
+The flag `--subresource` is added to get, patch, edit and replace with kubectl commands to 
```
Hi @divya-mohan0209
Can we write like this:
> The flag `--subresource`  is used with kubectl commands to perform fetch and update the `status` and `scale` subresources for all the API resources that support these subresources.

is it seems good to you?

divya-mohan0209 replied:
Wdyt? @sftim I'd like your two cents on this specific portion please? 

The flag `--subresource` is used with the kubectl get, patch, edit, and replace commands to fetch and update the subresources, `status` and `scale`, for all the API resources that support them.

Shubham82 replied:
> The flag --subresource is used with the kubectl get, patch, edit, and replace commands to fetch and update the subresources, status and scale, for all the API resources that support them.

Thanks, it's better than mine.
let's wait for @sftim response.

Shubham82 replied:
Hi @divya-mohan0209 
can I update this as per your comment, after that if there is any comment I will update it accordingly?

divya-mohan0209 replied:
Should be fine :)

Shubham82 replied:
Thanks!

Shubham82 replied:
addressed

Shubham82 commented:
I have addressed the comments.
PTAL!

tengqm commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -417,6 +417,86 @@ kubectl patch deployment patch-demo --patch-file patch-file.json
+
+{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
+
+The flag `--subresource=[subresource-name]` is used with kubectl commands like get, patch, edit and replace to fetch and update `status` and `scale` subresources of the resources (applicable for kubectl version v1.24 or more). This flag is used with all the API resources (built-in and CRs) which has `status` or `scale` subresource. Deployment is one of the examples which supports these subresourc ...
```
PLEASE WRAP THE LONG LINES, e.g.

```suggestion
The flag `--subresource=[subresource-name]` is used with kubectl commands like get, patch,
edit and replace to fetch and update `status` and `scale` subresources of the resources
(applicable for kubectl version v1.24 or more). This flag is used with all the API resources
(built-in and CRs) which has `status` or `scale` subresource. Deployment is one of the examples
which supports these subresources.
```

Shubham82 replied:
I will update this as per your comment soon.

Shubham82 replied:
addressed

tengqm commented on content/en/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md:
```diff
@@ -417,6 +417,86 @@ kubectl patch deployment patch-demo --patch-file patch-file.json
+
+Here's a manifest for a Deployment that has two replicas:
+
+{{< codenew file="application/deployment-subresource.yaml" >}}
```
Do we have to introduce a new YAML for this task?

Shubham82 replied:
Hi @tengqm, thanks for taking look at this.

>Do we have to introduce a new YAML for this task?

For every example/strategy used [In Update API Objects in Place Using kubectl patch](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/) task, there is a separate deployment YAML and has different configurations according to strategy. So for this, we can use a new deployment YAML that is related to this only so it will be more useful to the users/readers. it maintains the uniformity of the docs.

tengqm replied:
@Shubham82 I asked this question because I failed to find anything special in this `deployment-subresource.yaml` file. In fact, it can be any generic Deployment, right? This was not meant to be a blocker for this PR. My intent is to make sure that we are adding new example files for good reasons. We have [a test case for all examples](https://github.com/kubernetes/website/blob/main/content/en/examples/examples_test.go). I'm not convinced that a new file should be added for this scenario. Thoughts?

Shubham82 replied:
Hi @tengqm
I agree with your point, it is a generic Deployment, but we can't use the deployment mentioned in the task. if the same (kind of) generic deployment is already used in the docs we can use that one also. could you please point me to that deployment I will test and update the content accordingly to that? WDYT?

tengqm replied:
Here are some existing Deployment manifests:

- content/en/examples/application/deployment.yaml
- content/en/examples/application/deployment-scale.yaml
- content/en/examples/application/simple_deployment.yaml 
- content/en/examples/application/update_deployment.yaml 

They are pretty much the same thing repeated again and again.

Shubham82 replied:
I have updated it. I used content/en/examples/application/deployment.yaml manifest for this particular task.
PTAL

Shubham82 commented:
Addressed the comments!
PTAL!

Shubham82 commented:
> @Shubham82 Please help mark comments *addressed" if you agree to the suggestion and you have revised the PR accordingly. For comments still under discussion, you can leave them open. This will help the next reviewer focusing on comments that are not addressed.

Sure I will comment on the suggestion which has been resolved.

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+these subresources. It is applicable for kubectl version v1.24 or more. Previously fetching 
+subresources (like status) of API objects via kubectl involved using `kubectl --raw`, and 
+patching subresources using kubectl is not possible at all and requires using curl directly, 
+this method was not user-friendly. So from now, onwards `kubectl` tool can be used to edit or 
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
-The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
+The flag `--subresource` is used with the kubectl get, patch, edit, and replace commands to
+fetch and update the subresources, `status` and `scale`, for all the API resources that
+support them. It is applicable for kubectl version v1.24 or higher. Previously fetching
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+The flag `--subresource` is used with the kubectl get, patch, edit, and replace commands to
+fetch and update the subresources, `status` and `scale`, for all the API resources that
+support them. It is applicable for kubectl version v1.24 or higher. Previously fetching
+subresources (like status) of API objects via kubectl involved using `kubectl --raw`, and
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+support them. It is applicable for kubectl version v1.24 or higher. Previously fetching
+subresources (like status) of API objects via kubectl involved using `kubectl --raw`, and
+patching subresources using kubectl was not possible at all and required using curl directly.
+This was not user-friendly. From v1.24, onwards `kubectl` tool can be used to edit or
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+subresources (like status) of API objects via kubectl involved using `kubectl --raw`, and
+patching subresources using kubectl was not possible at all and required using curl directly.
+This was not user-friendly. From v1.24, onwards `kubectl` tool can be used to edit or
+patch the `status` subresource on a CRD. See [How to patch deployment by using subresource flag](/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#scale-kubectl-patch).
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
+patch the `status` subresource on a CRD. See [How to patch deployment by using subresource flag](/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#scale-kubectl-patch).
 
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+This page is part of the documentation for Kubernetes v{{< skew currentVersion >}}. If you are running a different version of Kubernetes, consult the documentation for that release.
```
addressed

Shubham82 commented on content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:
```diff
@@ -1053,13 +1053,18 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
-The easier way to patch the status subresource from the CLI is directly interacting with the API server using the `curl` tool, in this example:
+This page is part of the documentation for Kubernetes v{{< skew currentVersion >}}. If you are running a different version of Kubernetes, consult the documentation for that release.
+
+Here is an example of how to patch the `status` subresource for the CRD, using `kubectl`:
```
addressed

tengqm commented:
/label tide/merge-method-squash
/lgtm

sftim reviewed (commented):
Good to see this ready to merge.

/approve
